app/services/create_punches_service.rb

```diff
@@ -0,0 +1,24 @@
+class CreatePunchesService
+  class << self
+    def call(collection_of_punches, user)
+      days_to_delete = collection_of_punches.map { |punch| punch['from'] }
+
+      ActiveRecord::Base.transaction do
+        delete_punches_for_the_same_day(user, days_to_delete)
```

Comment:
why?

Reply:
I use this approach because is the one that is being used on the web.
On the mobile, you will select the date and create both punches, before and after lunch. So we need to guarantee that does not have any other punch on the same day.

Reply:
ok, the action is to create punch and not to delete them, if there's a collision  it should return a validation error

Reply:
But, that isn't an endpoint to delete punches, and for now, if the app creates a punch that is wrong it should stay until we create the delete endpoint?

Reply:
This delete was removed, instead if there are any punch on the same date, an error will be returned